services/management/dbaas/xtradb_cluster_service.go

```diff
@@ -0,0 +1,185 @@
+		},
+	}
+
+	out, err := s.controllerClient.ListXtraDBClusters(ctx, &in)
+	if err != nil {
+		return nil, err
+	}
+
+	clusters := []*dbaasv1beta1.ListXtraDBClustersResponse_Cluster{}
+	for _, c := range out.Clusters {
+		cluster := dbaasv1beta1.ListXtraDBClustersResponse_Cluster{
+			Name: c.Name,
+			Params: &dbaasv1beta1.XtraDBClusterParams{
+				ClusterSize: c.Params.ClusterSize,
+				Pxc: &dbaasv1beta1.XtraDBClusterParams_PXC{
+					ComputeResources: &dbaasv1beta1.ComputeResources{
+						CpuM:        c.Params.Pxc.ComputeResources.CpuM,
+						MemoryBytes: c.Params.Pxc.ComputeResources.MemoryBytes,
+					},
+				},
+				Proxysql: &dbaasv1beta1.XtraDBClusterParams_ProxySQL{
+					ComputeResources: &dbaasv1beta1.ComputeResources{
+						CpuM:        c.Params.Proxysql.ComputeResources.CpuM,
+						MemoryBytes: c.Params.Proxysql.ComputeResources.MemoryBytes,
+					},
+				},
+			},
+		}
+
+		clusters = append(clusters, &cluster)
+	}
+
+	return &dbaasv1beta1.ListXtraDBClustersResponse{Clusters: clusters}, nil
+}
+
+// CreateXtraDBCluster creates XtraDB cluster with given parameters.
+//nolint:dupl
+func (s XtraDBClusterService) CreateXtraDBCluster(ctx context.Context, req *dbaasv1beta1.CreateXtraDBClusterRequest) (*dbaasv1beta1.CreateXtraDBClusterResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	in := dbaascontrollerv1beta1.CreateXtraDBClusterRequest{
+		KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
+			Kubeconfig: kubernetesCluster.KubeConfig,
+		},
+		Name: req.Name,
+		Params: &dbaascontrollerv1beta1.XtraDBClusterParams{
+			ClusterSize: req.Params.ClusterSize,
+			Pxc: &dbaascontrollerv1beta1.XtraDBClusterParams_PXC{
+				ComputeResources: &dbaascontrollerv1beta1.ComputeResources{
+					CpuM:        req.Params.Pxc.ComputeResources.CpuM,
+					MemoryBytes: req.Params.Pxc.ComputeResources.MemoryBytes,
+				},
+			},
+			Proxysql: &dbaascontrollerv1beta1.XtraDBClusterParams_ProxySQL{
+				ComputeResources: &dbaascontrollerv1beta1.ComputeResources{
+					CpuM:        req.Params.Proxysql.ComputeResources.CpuM,
+					MemoryBytes: req.Params.Proxysql.ComputeResources.MemoryBytes,
+				},
+			},
+		},
+	}
+
+	_, err = s.controllerClient.CreateXtraDBCluster(ctx, &in)
+	if err != nil {
+		return nil, err
+	}
+
+	return &dbaasv1beta1.CreateXtraDBClusterResponse{}, nil
+}
+
+// UpdateXtraDBCluster updates XtraDB cluster.
+//nolint:dupl
+func (s XtraDBClusterService) UpdateXtraDBCluster(ctx context.Context, req *dbaasv1beta1.UpdateXtraDBClusterRequest) (*dbaasv1beta1.UpdateXtraDBClusterResponse, error) {
+	kubernetesCluster, err := models.FindKubernetesClusterByName(s.db.Querier, req.KubernetesClusterName)
+	if err != nil {
+		return nil, err
+	}
+
+	in := dbaascontrollerv1beta1.UpdateXtraDBClusterRequest{
+		KubeAuth: &dbaascontrollerv1beta1.KubeAuth{
+			Kubeconfig: kubernetesCluster.KubeConfig,
+		},
+		Name: req.Name,
```

Comment:
At line 130 we are getting the cluster by name and here we are re-setting the name but i makes me think?
1) Is it possible to rename a cluster?
2) What happens is `req.Name` is a duplicated name?

Reply:
As you can see `UpdateXtraDBCluster` is not implemented yet - so it is too early fo fix here anything.
https://github.com/percona-platform/dbaas-controller/blob/main/service/cluster/xtra_db_cluster.go#L128-L131